Formula/pmtiles.rb

```diff
@@ -0,0 +1,17 @@
+class Pmtiles < Formula
+  desc "Single-file executable tool for creating, reading and uploading PMTiles archives"
+  homepage "https://protomaps.com/docs/pmtiles"
+  url "https://github.com/protomaps/go-pmtiles/archive/refs/tags/v1.7.0.tar.gz"
+  sha256 "b97095cccba9d13d79f52e216ed25d598f2fc203e050908ca19e06ff8060f2d7"
+  license "BSD-3-Clause"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", "#{bin}/pmtiles"
+  end
+
+  test do
+    shell_output("#{bin}/pmtiles --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.